src/storage/index/LookupBaseProcessor.inl

```diff
@@ -22,8 +22,20 @@ cpp2::ErrorCode LookupBaseProcessor<REQ, RESP>::requestCheck(const cpp2::LookupI
     planContext_->isEdge_ = indices.get_is_edge();
     if (planContext_->isEdge_) {
         planContext_->edgeType_ = indices.get_tag_or_edge_id();
+        auto edgeName = this->env_->schemaMan_->toEdgeName(spaceId_, planContext_->edgeType_);
+        if (!edgeName.ok()) {
+            return cpp2::ErrorCode::E_EDGE_NOT_FOUND;
+        }
+        planContext_->edgeName_ = std::move(edgeName.value());
+        schema_ = this->env_->schemaMan_->getEdgeSchema(spaceId_, planContext_->edgeType_);
     } else {
         planContext_->tagId_ = indices.get_tag_or_edge_id();
+        auto tagName = this->env_->schemaMan_->toTagName(spaceId_, planContext_->tagId_);
+        if (!tagName.ok()) {
+            return cpp2::ErrorCode::E_TAG_NOT_FOUND;
+        }
+        planContext_->tagName_ = std::move(tagName.value());
+        schema_ = this->env_->schemaMan_->getTagSchema(spaceId_, planContext_->tagId_);
```

Comment:
The judgment of if (schema_ == nullptr) should be here
 not in line 243

Reply:
That's because `schmea_` is not used in all cases, only when we need to read data.

Reply:
I understand what you mean, not all schama is used
Only `buildPlanWithData ` and `buildPlanWithDataAndFilter `   use `schema `to read data
But as you said, there are two small problems:
1) The schema is null or not, it needs to be judged in at least two places
2) `buildPlanBasic `and `buildPlanWithFilter ` only fetch and read data from the index, but when the schema does not exist, the index cannot exist

so the schema is null or not can be unified ?